public/app/plugins/panel/annolist/AnnoListPanel.test.tsx

```diff
@@ -0,0 +1,265 @@
+import React from 'react';
+import { render, screen, waitFor } from '@testing-library/react';
+
+import { AnnoListPanel, Props } from './AnnoListPanel';
+import { AnnotationEvent, FieldConfigSource, getDefaultTimeRange, LoadingState } from '@grafana/data';
+import { AnnoOptions } from './types';
+import { backendSrv } from '../../../core/services/backend_srv';
+import userEvent from '@testing-library/user-event';
+import { silenceConsoleOutput } from '../../../../test/core/utils/silenceConsoleOutput';
+
+jest.mock('@grafana/runtime', () => ({
+  ...((jest.requireActual('@grafana/runtime') as unknown) as object),
+  getBackendSrv: () => backendSrv,
+}));
+
+jest.mock('app/features/dashboard/services/DashboardSrv', () => ({
+  getDashboardSrv: () => ({
+    getCurrent: () => ({ id: 1, formatDate: (time: number) => new Date(time).toISOString() }),
+  }),
+}));
```

Comment:
@hugohaggmark I guess we both struggle with this kind of mocks. Personally - I hate them 😬 

In the routing work, I have decided to use service setters rather than mocks, as it closer resembles the app setup than this criptic mock. And this mock will fail as soon as we change the import path for example. 

See what I did here: https://github.com/grafana/grafana/blob/master/public/app/features/dashboard/state/initDashboard.test.ts#L139 and here https://github.com/grafana/grafana/blob/master/public/app/features/dashboard/services/DashboardLoaderSrv.ts#L144

I think the setter pattern is so much easier to test and reson about, and maybe we should establish this as service dependency testing patter, what do you think?

Reply:
I totally agree, and I'll change this.

Reply:
I think I'll bring that to a weekly and we can establish this as a recommended way to use services dependencies in tests